app/src/main/java/com/example/atheneum/fragments/AddBookFragment.java

```diff
@@ -173,6 +173,10 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                             imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
                             mainActivity.getSupportFragmentManager().beginTransaction().remove(AddBookFragment.this).commit();
 
+                            // return to home page fragment
+                            mainActivity.getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, new HomeFragment()).commit();
+
```

Comment:
I think this is okay for now but I think it would be better to return to the owner activity. I know you had a hard time placing where adding books should show up in the UI but I think the best place to (eventually) put it would be in the owner fragment. The owner fragment could default to showing the list of all books the owner owns. In that fragment you can have a floating action bar with a "+" symbol that when clicked will either inflate this fragment (AddBookFragment) on top of the MainActivity or spawn a separate activity containing the AddBookFragment. This way you could remove adding books from the nav bar since it's kind of awkward being there right now.